jgrapht-core/src/main/java/org/jgrapht/alg/scoring/PageRank.java

```diff
@@ -0,0 +1,397 @@
+ * @author Dimitrios Michail
+ * @since August 2016
+ */
+public final class PageRank<V, E>
+    implements VertexScoringAlgorithm<V, Double>
+{
+
+    /**
+     * Default number of maximum iterations.
+     */
+    public static final int MAX_ITERATIONS_DEFAULT = 100;
+
+    /**
+     * Default value for the tolerance. The calculation will stop if the
+     * difference of PageRank values between iterations change less than this
+     * value.
+     */
+    public static final double TOLERANCE_DEFAULT = 0.0001;
+
+    /**
+     * Damping factor default value.
+     */
+    public static final double DAMPING_FACTOR_DEFAULT = 0.85d;
+
+    private final Graph<V, E> g;
+    private Map<V, Double> scores;
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     */
+    public PageRank(Graph<V, E> g)
+    {
+        this(
+            g,
+            DAMPING_FACTOR_DEFAULT,
+            MAX_ITERATIONS_DEFAULT,
+            TOLERANCE_DEFAULT,
+            false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     */
+    public PageRank(Graph<V, E> g, double dampingFactor)
+    {
+        this(
+            g,
+            dampingFactor,
+            MAX_ITERATIONS_DEFAULT,
+            TOLERANCE_DEFAULT,
+            false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public PageRank(Graph<V, E> g, double dampingFactor, int maxIterations)
+    {
+        this(g, dampingFactor, maxIterations, TOLERANCE_DEFAULT, false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of PageRank
+     *        values between iterations change less than this value
+     */
+    public PageRank(
+        Graph<V, E> g,
+        double dampingFactor,
+        int maxIterations,
+        double tolerance)
+    {
+        this(g, dampingFactor, maxIterations, tolerance, false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * This constructor allows the caller to adjust all parameters as well as to
+     * use a weighted variant of the algorithm.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of PageRank
+     *        values between iterations change less than this value
+     * @param weighted assume the edges are weighted or not
+     */
```

Comment:
Several other algorithms have the similar issues when there exists a weighted and an unweighted version of algorithm. Thus far we've had the standard approach that in case an unweighted graph is provided, the unweighted version is run, otherwise the weighted version. If the user intends to run the unweighted version of the algorithm on a weighted graph, thereby ignoring the edge weights, the user has to encapsulate the graph using the GraphDelegator class AsUnweightedGraph. I think this approach may work here as well?